rollup.config.js

```diff
@@ -0,0 +1,65 @@
+import peerDepsExternal from 'rollup-plugin-peer-deps-external'
+import resolve from '@rollup/plugin-node-resolve'
+import commonjs from '@rollup/plugin-commonjs'
+import { babel } from '@rollup/plugin-babel'
+import fileSize from 'rollup-plugin-filesize'
+import typescript from 'rollup-plugin-typescript2'
+
+export default {
+  input: './src/lib.ts',
+  output: [
+    {
+      dir: 'dist/cjs/src',
+      preserveModules: true,
+      preserveModulesRoot: 'src',
+      format: 'cjs',
+      sourcemap: true,
+      exports: 'named'
+    },
+    {
+      dir: 'dist/esm/src',
+      preserveModules: true,
+      preserveModulesRoot: 'src',
+      format: 'esm',
+      sourcemap: true,
+      exports: 'named'
+    }
+  ],
+  plugins: [
+    peerDepsExternal(),
+    resolve(),
+    typescript({
+      tsconfig: './tsconfig.json',
+      useTsconfigDeclarationDir: true
+    }),
+    babel({
+      babelHelpers: 'bundled',
+      exclude: 'node_modules/**'
+    }),
+    commonjs(),
+    fileSize()
+  ],
+  external: [
```

Comment:
Should we potentially do `Object.keys(require("./package.json").peerDependencies)` here? Let's also review the peer list—IMO a few of them can no longer be peer, e.g. `downshift` and `lodash`.